src/app/(auth)/sign-up/_component/SignupForm.tsx

```diff
@@ -83,6 +85,16 @@ export default function SignupForm() {
     setValue('birth', formattedValue, { shouldValidate: true });
   };
 
+  const handleClickAgreement = ({ first, second }: { first?: boolean; second?: boolean }) => {
+    if (first) {
+      setIsAgreementOpen((agree) => [!agree[0], agree[1]]);
+    }
+
+    if (second) {
+      setIsAgreementOpen((agree) => [agree[0], !agree[1]]);
+    }
+  };
```

Comment:
first,second보단 index로 관리하면 코드 더 깔끔해질 것 같아요! 이런 식으로?
```suggestion
  const handleClickAgreement = (index: number) => {
    setIsAgreementOpen((agreement) => agreement.map((isAgree, i) => i === index ? !isAgree: isAgree));
  };
```

근데 개인적으로는 isAgreementOpen을 배열보다는 객체로 관리하는 것은 어떨까요?

```suggestion
  const [isAgreementOpen, setIsAgreementOpen] = useState({first: false, second: false})
  
  const handleClickAgreement = (key: 'first' | 'second') {
    setIsAgreementOpen((prev) => ({...prev, [key]: !prev[key]}));
  }
```